app/services/alerts_service.rb

```diff
@@ -0,0 +1,78 @@
+class AlertsService
+  def initialize; end
+
+  def call
+    search_active_alerts
+  end
+
+  private
+
+  def search_active_alerts
+    Alert.find_each do |alert|
+      last_sent_date = alert.last_sent_date
+
+      next unless last_sent_date.nil? || Time.zone.today > last_sent_date + alert.frequency.days
+
+      success_rate = alert_metric_entity(alert)
+
+      search_below_rate(alert, success_rate) if alert.active & success_rate
+    end
+  end
+
+  def alert_metric_entity(alert)
+    repository = alert.repository
+    department = alert.department
+
+    entity_is_repo = repository.present?
+
+    return unless entity_is_repo || department.present?
+
+    entity_name = entity_is_repo ? Repository.name : Department.name
```

Comment:
maybe `repository.name : department.name` with lowercase?

Reply:
I've tested and it does not work 😩